bioptim/optimization/optimal_control_program.py

```diff
@@ -820,7 +821,9 @@ def solve(
             from ..interfaces.acados_interface import AcadosInterface
 
             if solver_options is None:
-                solver_options = {}
+                solver_options = SolverOptionsACADOS().__dict__
+            else:
+                solver_options = solver_options.__dict__
```

Comment:
I cannot comment later, but there is another set of options (while `is_warm_starting`). They should be moved to your new class!